pkg/sdk/datatypes/table.go

```diff
@@ -23,17 +29,79 @@ func (c *TableDataTypeColumn) ColumnType() DataType {
 }
 
 func (t *TableDataType) ToSql() string {
-	return t.underlyingType
+	columns := strings.Join(collections.Map(t.columns, func(col TableDataTypeColumn) string {
+		return fmt.Sprintf("%s %s", col.name, col.dataType.ToSql())
+	}), ", ")
+	return fmt.Sprintf("%s (%s)", TableLegacyDataType, columns)
 }
 
 func (t *TableDataType) ToLegacyDataTypeSql() string {
-	return TableLegacyDataType
+	columns := strings.Join(collections.Map(t.columns, func(col TableDataTypeColumn) string {
+		return fmt.Sprintf("%s %s", col.name, col.dataType.ToLegacyDataTypeSql())
+	}), ", ")
+	return fmt.Sprintf("%s (%s)", TableLegacyDataType, columns)
 }
 
 func (t *TableDataType) Canonical() string {
-	return TableLegacyDataType
+	columns := strings.Join(collections.Map(t.columns, func(col TableDataTypeColumn) string {
+		return fmt.Sprintf("%s %s", col.name, col.dataType.Canonical())
+	}), ", ")
+	return fmt.Sprintf("%s (%s)", TableLegacyDataType, columns)
 }
 
 func (t *TableDataType) Columns() []TableDataTypeColumn {
 	return t.columns
 }
+
+func parseTableDataTypeRaw(raw sanitizedDataTypeRaw) (*TableDataType, error) {
+	r := strings.TrimSpace(strings.TrimPrefix(raw.raw, raw.matchedByType))
+	if r == "()" {
+		return &TableDataType{
+			columns:        make([]TableDataTypeColumn, 0),
+			underlyingType: raw.matchedByType,
+		}, nil
+	}
+	if r == "" || (!strings.HasPrefix(r, "(") || !strings.HasSuffix(r, ")")) {
+		logging.DebugLogger.Printf(`table %s could not be parsed, use "%s(argName argType, ...)" format`, raw.raw, raw.matchedByType)
+		return nil, fmt.Errorf(`table %s could not be parsed, use "%s(argName argType, ...)" format`, raw.raw, raw.matchedByType)
+	}
+	onlyArgs := r[1 : len(r)-1]
+	columns, err := collections.MapErr(strings.Split(onlyArgs, ","), func(arg string) (TableDataTypeColumn, error) {
+		argParts := strings.Split(strings.TrimSpace(arg), " ")
```

Comment:
we cant like this because of eg. `DOUBLE PRECISION` -> let's split here on the first space index instead